python/pyspark/sql/tests/test_catalog.py

```diff
@@ -210,17 +210,34 @@ def test_list_columns(self):
 
     def test_table_exists(self):
         # SPARK-36176: testing that table_exists returns correct boolean
+        from pyspark.sql.catalog import Table
```

Comment:
Let's leave the tests as is since the doctests cover it - I think it's too much

Reply:
@HyukjinKwon Actually, the last test here currently fails, which is why I added it. The test shows that while a temporary view gets correctly reported in listTables, it doesn't return true using tableExists. Now, I'm not sure if there's someone on the scala side who might have more insights into why this is happening and/or whether the scala documentation is outdated stating that temporary views are ok for tableExists

Reply:
Added Scala tests showing the issue

Reply:
Moved all tests as you suggested.